### PR TITLE
add some comments for SeriesReader

### PR DESCRIPTION
**Pull request by mychaow**
### Content3 ...

<hr>

This PR has:
- [ ] been self-reviewed.
    - [ ] concurrent read
    - [ ] concurrent write
    - [ ] concurrent read and write 
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. 
- [ ] added or updated version, __license__, or notice information
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious 
  for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold 
  for code coverage.
- [ ] added integration tests.
- [ ] been tested in a test IoTDB cluster.

<hr>

### CODE REVIEW

**Comment by jixuan1989** [comment]
what is this PR for?

**Comment by mychaow** [comment]
> what is this PR for?

just add some comments for SeriesReader

**Comment by weizihan0110** [comment]
Comments on codes that describe "what" and "why" are so welcome!

File: server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java [hunk 1]
```diff
@@ -390,15 +390,36 @@ boolean hasNextPage() throws IOException {
...
-        unpackAllOverlappedTimeSeriesMetadataToCachedChunkMetadata(endpointTime, false);
-        unpackAllOverlappedChunkMetadataToCachedPageReaders(endpointTime, false);
+      unpackFirstPageReader();
+    }
+
+    if (isExistOverlappedPage()) {
+      return true;
+    }
+
+    // make sure firstPageReader won't be null while cachedPageReaders has more cached page readers
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java:400]:
```suggestion
    // make sure firstPageReader won't be null while the cachedPageReaders has more cached page readers
```

File: server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java [hunk 2]
```diff
@@ -601,31 +599,46 @@ private boolean hasNextOverlappedPage() throws IOException {
...
           /*
-           * get current first point in mergeReader, this maybe overlapped latter
+           * get current first point in mergeReader, this maybe overlapped later
            */
           TimeValuePair timeValuePair = mergeReader.currentTimeValuePair();
 
           if (orderUtils.isExcessEndpoint(timeValuePair.getTimestamp(), currentPageEndPointTime)) {
+            /*
+             * when the merged point excesses the currentPageEndPointTime, we have read all overlapped data before currentPageEndPointTime
+             * 1. has cached batch data, we need not to read more data, just use the cached data later
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java:609]:
```suggestion
             * 1. has cached batch data, we don't need to read more data, just use the cached data later
```

File: server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java [hunk 3]
```diff
@@ -601,31 +599,46 @@ private boolean hasNextOverlappedPage() throws IOException {
...
+             * 2. has first page reader, means first page reader last endTime < currentTimeValuePair.getTimestamp(),
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java:610]:
```suggestion
             * 2. has first page reader, which means first page reader last endTime < currentTimeValuePair.getTimestamp(),
```

File: server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java [hunk 4]
```diff
@@ -601,31 +599,46 @@ private boolean hasNextOverlappedPage() throws IOException {
...
+             * we could just use the first page reader later
+             * 3. sequence page reader is not empty,  means first page reader last endTime < currentTimeValuePair.getTimestamp(),
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java:612]:
```suggestion
             * 3. sequence page reader is not empty, which means first page reader last endTime < currentTimeValuePair.getTimestamp(),
```

File: server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java [hunk 5]
```diff
@@ -601,31 +599,46 @@ private boolean hasNextOverlappedPage() throws IOException {
...
+             * we could use the first sequence page reader later
+             */
             if (cachedBatchData.hasCurrent() || firstPageReader != null || !seqPageReaders
                 .isEmpty()) {
               break;
             }
+            // so, we does not has other data except mergeReader
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java:619]:
```suggestion
            // so, we don't have other data except mergeReader
```

File: server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java [hunk 6]
```diff
@@ -601,31 +599,46 @@ private boolean hasNextOverlappedPage() throws IOException {
...
           }
 
+          // unpack all overlapped data for the first timeValuePair
           unpackAllOverlappedTsFilesToTimeSeriesMetadata(timeValuePair.getTimestamp());
           unpackAllOverlappedTimeSeriesMetadataToCachedChunkMetadata(
               timeValuePair.getTimestamp(), false);
           unpackAllOverlappedChunkMetadataToCachedPageReaders(timeValuePair.getTimestamp(), false);
           unpackAllOverlappedUnseqPageReadersToMergeReader(timeValuePair.getTimestamp());
 
+          // from now, the unsequence reader is all unpacked, so we need not to consider it
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java:630]:
```suggestion
          // from now, the unsequence reader is all unpacked, so we don't need to consider it
```

File: server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java (hunk 1 above)
Comment (mychaow) [reply server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java:400]: Thanks for detailed review.

**Review by HTHou — APPROVED** [review]

**Pull request merged**